src/udev/udev-builtin-blkid.c

```diff
@@ -321,8 +321,8 @@ static int builtin_blkid(struct udev_device *dev, int argc, char *argv[], bool t
         if (is_gpt)
                 find_gpt_root(dev, pr, test);
 
-        blkid_free_probe(pr);
 out:
+        blkid_free_probe(pr);
```

Comment:
By the way, the leak was introduced by d47f6ca5f9b7a0b400d8b, which was also mentioned in #6110 . @g2p do you know how d47f6ca5f9b7a0b400d can cause #6110?